lib/name.mli

```diff
@@ -27,11 +27,20 @@ open Cstruct
 type label
 
 (** Domain name, as a list of strings. *)
-type domain_name = string list
+type domain_name
 
 (** Domain name map *)
 module Map: Map.S with type key = domain_name
 
+(** Create null domain name *)
+val empty: domain_name
+
+(** Render a {! domain_name} to a {! string} list. *)
+val domain_name_to_string_list : domain_name -> string list
+
+(** Convert a standard domain {! string} list to a {! domain_name}. *)
+val string_to_domain_name : string list -> domain_name
+
 (** Render a {! domain_name} to a simple string. *)
 val domain_name_to_string : domain_name -> string
```

Comment:
i naively tried this, noticed how pervasive ``open Cstruct`` was gave up :)
commit 17f69f5fdc155b878d03111cbc4353c31fdf06c7
revert 1593fa76b606405a7842dffe693e2e169f1fa9f5